test/types/object.js

```diff
@@ -2163,6 +2163,33 @@ describe('object', () => {
 
     });
 
+    describe('ES6 Classes', () => {
+
+        it('should differentiate between ES6 classes and functions', (done) => {
+
+            const classSchema = Joi.object({
+                _class: Joi.func().class()
+            });
+
+            const testFunc = function () {};
+            const testClass = class MyClass {};
+
+            classSchema.validate({ _class: testFunc }, (err, invalidValue) => {
```

Comment:
Can we not use the Helper that we have just like in the other tests?

Also, can we add some additional negative test cases such as `undefined`, `null`, etc. It looks like there are some cases in the current code where we could throw an error instead of actually responding with a Joi error.

Reply:
Sure thing, updated!